bundles/config/org.eclipse.smarthome.config.discovery.usbserial.linux.sysfs/src/main/java/org/eclipse/smarthome/config/discovery/usbserial/linux/sysfs/internal/SysfsUsbSerialScanner.java

```diff
@@ -0,0 +1,265 @@
+ * <p/>
+ * A scan starts by inspecting the contents of the directory '/sys/class/tty'. This directory contains a symbolic link
+ * for every serial port style device that points to the device information provided by the sysfs in some subdirectory
+ * of '/sys/devices'.
+ * <p/>
+ * The scan considers only those serial ports for which the corresponding device file (in folder '/dev'; e.g.:
+ * '/dev/ttyUSB0') is both readable and writable, as otherwise the serial port cannot be used by any binding. For those
+ * serial ports, the scan checks whether the serial port actually originates from a USB device, by inspecting the
+ * information provided by the sysfs in the folder pointed to by the symbolic link.
+ * <p/>
+ * If the device providing the serial port is a USB device, information about the device (vendor ID, product ID, etc.)
+ * is collected from the sysfs and returned together with the name of the serial port in form of a
+ * {@link UsbSerialDeviceInformation}.
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+@Component(configurationPid = "discovery.usbserial.linux.sysfs.usbserialscanner")
+@NonNullByDefault
+public class SysfsUsbSerialScanner implements UsbSerialScanner {
+
+    private final Logger logger = LoggerFactory.getLogger(SysfsUsbSerialScanner.class);
+
+    public static final String SYSFS_TTY_DEVICES_DIRECTORY_ATTRIBUTE = "sysfsTtyDevicesPath";
+    public static final String DEV_DIRECTORY_ATTRIBUTE = "devPath";
+
+    private static final String SYSFS_TTY_DEVICES_DIRECTORY_DEFAULT = "/sys/class/tty";
+    private static final String DEV_DIRECTORY_DEFAULT = "/dev";
+
+    private String sysfsTtyDevicesDirectory = SYSFS_TTY_DEVICES_DIRECTORY_DEFAULT;
+    private String devDirectory = DEV_DIRECTORY_DEFAULT;
+
+    private static final String SYSFS_FILENAME_USB_VENDOR_ID = "idVendor";
+    private static final String SYSFS_FILENAME_USB_PRODUCT_ID = "idProduct";
+    private static final String SYSFS_FILENAME_USB_SERIAL_NUMBER = "serial";
+    private static final String SYSFS_FILENAME_USB_MANUFACTURER = "manufacturer";
+    private static final String SYSFS_FILENAME_USB_PRODUCT = "product";
+
+    /**
+     * In the sysfs, directories for USB interfaces have the following format (cf., e.g.,
+     * http://www.linux-usb.org/FAQ.html#i6), where there can be one or more USB port numbers separated by dots.
+     *
+     * <pre>
+     * {@code <#bus>-<#port>.<#port>:<#config>:<#interface>}
+     * </pre>
+     *
+     * Example: {@code 3-1.3:1.0}
+     * <p/>
+     * This format is captured by this {@link Pattern}.
+     */
+    private static final Pattern SYSFS_USB_INTERFACE_DIRECTORY_PATTERN = Pattern
+            .compile("\\d+-(\\d+\\.?)*\\d+:\\d+\\.\\d+");
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        if (config.containsKey(SYSFS_TTY_DEVICES_DIRECTORY_ATTRIBUTE)) {
+            sysfsTtyDevicesDirectory = ObjectUtils.toString(config.get(SYSFS_TTY_DEVICES_DIRECTORY_ATTRIBUTE));
```

Comment:
`sysfsTtyDevicesDirectory` is set twice if `SYSFS_TTY_DEVICES_DIRECTORY_ATTRIBUTE` exists in the configuration. You can also consider using 

```java
config.getOrDefault(SYSFS_TTY_DEVICES_DIRECTORY_ATTRIBUTE, SYSFS_TTY_DEVICES_DIRECTORY_DEFAULT);
```

Reply:
As just discussed with @amitjoy: there should not be a problem with setting `sysfsTtyDevicesDirectory` twice; however, the configuration should not only be set in the `modified` method, but also on activation of the component. I will add this to the code. Thanks for the discussion, @amitjoy .